Add more documentation with technical details about how the new website is generated (#2611)
google/android-fhir: docs/contrib/docs.md

1. Confirm Codespace's _"Port Forward"_ prompt to open `https://...app.github.dev`, et voilà!

## Implementation Details

The website is built using [Material](https://squidfunk.github.io/mkdocs-material/) for [MkDocs](https://www.mkdocs.org/).

> The [`Build` GitHub Action](https://github.com/google/android-fhir/actions/workflows/build.yml)
> (see [`build.yml`](https://github.com/google/android-fhir/blob/master/.github/workflows/build.yml)) has the required steps
> which runs the [`build-docs.bash`](https://github.com/google/android-fhir/blob/master/build-docs.bash) script which generates the HTML in the `site/` directory.
> That `site/` directory is intentionally on `.gitignore` and only created dynamically during the build process, but not commited into the Git repository.

Reviewer: ```suggestion
The GitHub Action [`Build`](https://github.com/google/android-fhir/actions/workflows/build.yml) (defined in [`build.yml`](https://github.com/google/android-fhir/blob/master/.github/workflows/build.yml)) runs the [`build-docs.bash`](https://github.com/google/android-fhir/blob/master/build-docs.bash) script to generate the HTML in the `site/` directory.
That `site/` directory is created dynamically during the build process but never commited into the Git repository since it is included in the `.gitignore` file.
```

> The basic docs related build steps intentionally runs for every PR, which helps to detect broken doc before merge (AKA "pre-submit"). The PRs do however obviously NOT actually update
> the live <https://google.github.io/android-fhir/> website; that of course only happens for builds of the default `master` branch! (The way this works technically is due to the
> automagical combination of `on: push: branches: [master]` and `if: ${{ github.event_name == 'push' }}` in `build.yml`.)

Reviewer: ```suggestion
The docs related build steps run for every PR, to help detect broken doc before merge. However, these runs do not update
the live website, which only happens for builds of the `master` branch (notice `on: push: branches: [master]` and `if: ${{ github.event_name == 'push' }}` in [`build.yml`](https://github.com/google/android-fhir/blob/master/.github/workflows/build.yml).)
```

> automagical combination of `on: push: branches: [master]` and `if: ${{ github.event_name == 'push' }}` in `build.yml`.)

Reviewer: can you also talk about how the generated html is deployed? i think it's by the `deploy-website` step in the build.yml file?

Reviewer: actually i'm curious about the difference between actions/upload-pages-artifact@v3 and actions/deploy-pages@v4

Author: Done, hope this helps?
